### PR TITLE
feat: add aa-signers package

### PR DESCRIPTION
**Pull request by avasisht23**
# Pull Request Checklist

- [x] Did you add new tests and confirm existing tests pass? (`yarn test`)
- [x] Did you update relevant docs? (docs are found in the `site` folder, see guidleines below)
- [x] Do your commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [x] Does your PR title also follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [x] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples? (e.g. `feat!: breaking change`)
- [x] Did you run lint (`yarn lint:check`) and fix any issues? (`yarn lint:fix`)
- [x] Is the base branch you're merging into `development` and not `main`?

---

### Focus of this PR:
This PR focuses on upgrading the Node version, updating dependencies, adding new files, and making changes to existing files in the `@alchemy/aa-signers` package.

### Detailed summary:
- Upgraded Node version from 18.10 to 18.16
- Added `tsconfig.json` file
- Updated `deploy-site.yml` and `publish-package.yml` workflows to use Node version 18.16
- Added `tsconfig.build.json` file with updated compiler options
- Updated content and links in `index.md` and `contributing.md` files
- Added new files: `signer.test.ts`, `vitest.config.ts`, `vitest.config.e2e.ts`, `getting-started.md`, `index.md`, `config.ts`, `package.json`, `package-overview.md`
- Updated content in `README.md` and `package.json` files

> The following files were skipped due to too many changes: `site/overview/package-overview.md`, `packages/signers/CHANGELOG.md`, `yarn.lock`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by avasisht23** [comment]
Current dependencies on/for this PR:
* development
  * **PR #227** <a href="https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/227" target="_blank">Graphite</a> 
    * **PR #228** <a href="https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/228" target="_blank">Graphite</a>  👈
      * **PR #229** <a href="https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/229" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/228?utm_source=stack-comment).

**Review by moldy530 — APPROVED** [review]

**Review by denniswon — APPROVED** [review]
Follow up task to be made for aa-signers documentation

File: packages/signers/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,168 @@
+# Change Log
+
+All notable changes to this project will be documented in this file.
+See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
+
+# 0.2.0 (2023-11-03)
```
Comment (denniswon) [inline packages/signers/CHANGELOG.md:6]: we are cutting a new release for light account factory update and sepolia arb support update, as v0.3.0. Think we should merge this in afterwards. cc: @moldy530

Comment (avasisht23) [reply packages/signers/CHANGELOG.md:6]: agree!

**Pull request merged**